### PR TITLE
(spanner): write and read array of bool, int64 and float64 types examples

### PR DESCRIPTION
**Pull request by jiren**
Example list
1. write read bool array
2. write read empty int64 array
3. write read null int64 array
4. write read int64 array
5. write read empty float64 array
6. write read null float64 array
7. write read float64 array

Ref issue: [#2579](https://github.com/googleapis/google-cloud-ruby/issues/2579)

### CODE REVIEW

File: spanner/spanner_samples.rb [hunk 1]
```diff
@@ -1223,6 +1234,161 @@ def query_with_timestamp project_id:, instance_id:, database_id:
...
+end
+
+def write_read_null_float64_array project_id:, instance_id:, database_id:
+  # [START spanner_write_read_null_float64_array]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
```
Comment (TheRoyalTnetennba) [inline spanner/spanner_samples.rb]: `require "securerandom"`

File: spanner/spanner_samples.rb [hunk 2]
```diff
@@ -1223,6 +1234,161 @@ def query_with_timestamp project_id:, instance_id:, database_id:
...
+end
+
+def write_read_float64_array project_id:, instance_id:, database_id:
+  # [START spanner_write_read_float64_array]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
```
Comment (TheRoyalTnetennba) [inline spanner/spanner_samples.rb]: `require "securerandom"`

File: spanner/spanner_samples.rb [hunk 3]
```diff
@@ -1223,6 +1234,167 @@ def query_with_timestamp project_id:, instance_id:, database_id:
...
+  client = spanner.client instance_id, database_id
+
+  box_id = SecureRandom.uuid
+  client.insert "Boxes", BoxId: box_id, ErrorChecks: [true, false, true]
+  results = client.read "Boxes", [:BoxId, :ErrorChecks], keys: box_id
+
+  results.rows.each do |row|
+    puts row["ErrorChecks"]
+  end
+
```
Comment (skuruppu) [inline spanner/spanner_samples.rb]:
```suggestion
```

File: spanner/spanner_samples.rb [hunk 4]
```diff
@@ -1223,6 +1234,167 @@ def query_with_timestamp project_id:, instance_id:, database_id:
...
+  client  = spanner.client instance_id, database_id
+
+  box_id = SecureRandom.uuid
+  client.insert "Boxes", BoxId: box_id, Heights: []
+  results = client.read "Boxes", [:BoxId, :Heights], keys: box_id
+
+  results.rows.each do |row|
+    puts row["Heights"].empty?
+  end
+
```
Comment (skuruppu) [inline spanner/spanner_samples.rb]:
```suggestion
```

File: spanner/spanner_samples.rb [hunk 5]
```diff
@@ -1223,6 +1234,167 @@ def query_with_timestamp project_id:, instance_id:, database_id:
...
+  client  = spanner.client instance_id, database_id
+
+  box_id = SecureRandom.uuid
+  client.insert "Boxes", BoxId: box_id, Heights: [nil, nil, nil]
+  results = client.read "Boxes", [:BoxId, :Heights], keys: box_id
+
+  results.rows.each do |row|
+    row["Heights"].each { |height| puts height.nil? }
+  end
+
```
Comment (skuruppu) [inline spanner/spanner_samples.rb]:
```suggestion
```

File: spanner/spanner_samples.rb [hunk 6]
```diff
@@ -1223,6 +1234,167 @@ def query_with_timestamp project_id:, instance_id:, database_id:
...
+  client  = spanner.client instance_id, database_id
+
+  box_id = SecureRandom.uuid
+  client.insert "Boxes", BoxId: box_id, Heights: [10, 11, 12]
+  results = client.read "Boxes", [:BoxId, :Heights], keys: box_id
+
+  results.rows.each do |row|
+    puts row["Heights"]
+  end
+
```
Comment (skuruppu) [inline spanner/spanner_samples.rb]:
```suggestion
```

File: spanner/spanner_samples.rb [hunk 7]
```diff
@@ -1223,6 +1234,167 @@ def query_with_timestamp project_id:, instance_id:, database_id:
...
+  client  = spanner.client instance_id, database_id
+
+  box_id = SecureRandom.uuid
+  client.insert "Boxes", BoxId: box_id, Weights: []
+  results = client.read "Boxes", [:BoxId, :Weights], keys: box_id
+
+  results.rows.each do |row|
+    puts row["Weights"].empty?
+  end
+
```
Comment (skuruppu) [inline spanner/spanner_samples.rb]:
```suggestion
```

File: spanner/spanner_samples.rb [hunk 8]
```diff
@@ -1223,6 +1234,167 @@ def query_with_timestamp project_id:, instance_id:, database_id:
...
+  client  = spanner.client instance_id, database_id
+
+  box_id = SecureRandom.uuid
+  client.insert "Boxes", BoxId: box_id, Weights: [nil, nil, nil]
+  results = client.read "Boxes", [:BoxId, :Weights], keys: box_id
+
+  results.rows.each do |row|
+    row["Weights"].each { |weight| puts weight.nil? }
+  end
+
```
Comment (skuruppu) [inline spanner/spanner_samples.rb]:
```suggestion
```

File: spanner/spanner_samples.rb [hunk 9]
```diff
@@ -1223,6 +1234,167 @@ def query_with_timestamp project_id:, instance_id:, database_id:
...
+  client  = spanner.client instance_id, database_id
+
+  box_id = SecureRandom.uuid
+  client.insert "Boxes", BoxId: box_id, Weights: [10.001, 11.1212, 104.4123101]
+  results = client.read "Boxes", [:BoxId, :Weights], keys: box_id
+
+  results.rows.each do |row|
+    puts row["Weights"]
+  end
+
```
Comment (skuruppu) [inline spanner/spanner_samples.rb]:
```suggestion
```

Comment (jiren) [reply spanner/spanner_samples.rb]: @skuruppu Removed new line from all the occurrences as mentioned above.

**Comment by skuruppu** [comment]
@jiren I didn't notice it earlier but this PR makes a modification to the `spanner_create_database` code sample. The change makes it look different to the code samples for other languages. Could you send me a follow-up PR that moves the change to add the boxes table in a different function that's not surrounded by the `spanner_create_database` tracking tag?

**Comment by jiren** [comment]
surer, @skuruppu I will create a followup PR.

**Pull request merged**